pallets/dao-assets/src/functions.rs

```diff
@@ -60,6 +70,42 @@ impl<T: Config> Pallet<T> {
 		Self::search_history(SupplyHistory::<T>::get(id), block)
 	}
 
+	pub fn remove_account_history(asset_id: T::AssetId, account: &T::AccountId) {
+		AccountHistory::<T>::remove_prefix((asset_id, account), None);
```

Comment:
`remove_prefix` takes limit as optional argument but `clear_prefix` takes limit as manditaory argument. Here we want to remove everything without any limit

Reply:
I know, but it's deprecated and will be removed soon. The reasoning of the core devs behind this (I guess at least) is that within blockchain each operation that has not constant complexity needs to have a definite upper bound. If there might be a trillion things to remove here, it might take even longer than the blocktime, hence the tx could never be included.

That's why it's explicit. The solution here is to set the limit to `MaxProposals + 1`.